app/controllers/cart_controller.rb

```diff
@@ -27,7 +27,7 @@ def update
       cart.add_quantity(params[:id]) unless cart.limit_reached?(params[:id])
     elsif params[:quantity] == "dec"
       cart.subtract_quantity(params[:id])
-      session[:cart].delete(params[:id]) if cart.quantity_zero?(params[:id])
+      return remove_item(params[:id]) if cart.quantity_zero?(params[:id])
```

Comment:
What is the point of this return?

Reply:
It was producing an error when I just tried to call `remove_item(params[:id])`